### PR TITLE
sycl::is_device_copyable tests implementation

### PR DESCRIPTION
**Pull request by AmirIpma**
Created tests for `sycl::is_device_copyable` type trait according to [test plan](https://github.com/KhronosGroup/SYCL-CTS/blob/SYCL-2020/test_plans/is_device_copyable.asciidoc)

### CODE REVIEW

File: tests/is_device_copyable/is_device_copyable_api.cpp [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+  if constexpr (std::is_same_v<T, char>) {
+    return 'c';
+  }
+  return 0;
+}
+
+TEST_CASE("is_device_copyable UnaryTrait requirements",
+          "[is_device_copyable]") {
+  using device_copyable_t = sycl::is_device_copyable<user_def_type>;
+  using integral_constant_t = std::integral_constant<bool, true>;
```
Comment (keryell) [inline tests/is_device_copyable/is_device_copyable_api.cpp:75]: Besides the fact I find your name unclear, you are redefining `std::true_type`.

File: tests/is_device_copyable/is_device_copyable_api.cpp [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+*******************************************************************************/
+
+#include "../common/common.h"
+#include "../common/type_list.h"
+
+#if SYCL_DEVICE_COPYABLE == 1
```
Comment (ProGTX) [inline tests/is_device_copyable/is_device_copyable_api.cpp:24]:
```suggestion
#if defined(SYCL_DEVICE_COPYABLE) && (SYCL_DEVICE_COPYABLE == 1)
```
From the spec:
>  If the implementation has this support, it must predefine the preprocessor macro SYCL_DEVICE_COPYABLE to 1, and it must not predefine this preprocessor macro if it does not have this support.

We could potentially also add a check that if the macro is defined, it must be defined to 1. But the above suggestion should be sufficient to make impls that don't support that pass this test.

File: tests/is_device_copyable/is_device_copyable_api.cpp [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+  const user_def_type reference_object(
+      expected_val<int>(), expected_val<float>(), expected_val<char>());
+
+  bool host_res;
+  user_def_type host_object(reference_object);
+  user_def_type device_object;
+  {
+    sycl::buffer<bool, 1> host_res_buf(&host_res, {1});
+    sycl::buffer<user_def_type, 1> host_data_buf(&host_object, {1});
+    sycl::buffer<user_def_type, 1> dev_data_buf(&device_object, {1});
```
Comment (ProGTX) [inline tests/is_device_copyable/is_device_copyable_api.cpp:119]:
```suggestion
    sycl::buffer<bool> host_res_buf(&host_res, {1});
    sycl::buffer<user_def_type> host_data_buf(&host_object, {1});
    sycl::buffer<user_def_type> dev_data_buf(&device_object, {1});
```

**Review by ProGTX — APPROVED** [review]

File: tests/is_device_copyable/is_device_copyable_api.cpp (hunk 2 above)
Comment (AmirIpma) [reply tests/is_device_copyable/is_device_copyable_api.cpp]: Applied.

File: tests/is_device_copyable/is_device_copyable_api.cpp (hunk 3 above)
Comment (AmirIpma) [reply tests/is_device_copyable/is_device_copyable_api.cpp]: I don't think it was really necessary, but applied.

**Review by keryell — APPROVED** [review]
Thanks.

**Pull request merged**